### DISCUSSION TITLE
Servers field in `openapi.json` not shown even after set `root_path`

### DISCUSSION THREAD

**Discussion by fahadh4ilyas** [fastapi/fastapi#6594] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.exceptions import HTTPException, RequestValidationError
from fastapi.responses import ORJSONResponse, RedirectResponse

app = FastAPI(
    root_path='/api/v1'
)
```

### Description

I'm trying to run my API behind proxy. The proxy I'm used is `apache2`. The API is running smoothly (I'm running the API using gunicorn). But, when I tried to open the swagger docs using `<domain>/api/v1/docs` and test one of the function, it's not working properly. And I realize that `servers` parameter in `openapi.json` is not shown. It should be like this:

```json
{
    "openapi": "3.0.2",
    "servers": [
        {"url": "/api/v1"}
    ],
    ...
}
```

But, what is shown is like this

```json
{
    "openapi": "3.0.2",
    ...
}
```

I think this is why when I test the API from the swagger docs, the test is not working properly.

### Operating System

Linux

### FastAPI Version

0.68.1

### Python Version

3.9.6

**Comment by ghandic** [comment]
You should add the `servers=` param into the `FastAPI` initialization, see the example in the unit tests

https://github.com/tiangolo/fastapi/blob/22528373bba6a654323de416ad5c867cbadb81bb/tests/test_openapi_servers.py

**Comment by fahadh4ilyas** [comment]
> You should add the `servers=` param into the `FastAPI` initialization, see the example in the unit tests
> 
> https://github.com/tiangolo/fastapi/blob/22528373bba6a654323de416ad5c867cbadb81bb/tests/test_openapi_servers.py

I believe from [documentation behind proxy](https://fastapi.tiangolo.com/advanced/behind-a-proxy/) telling that they are automatically added?

But, I already try it like this and still not working

```python
app = FastAPI(
    root_path='/api/v1',
    servers=[{'url': '/api/v1'}]
)
```

**Comment by ghandic** [comment]
Is it in the `openapi.json` now though? I haven't done much work with this problem, but have you tried dumbing it down all the way to the way it is documented and making onse change at a time back to your original solution?

**Comment by fahadh4ilyas** [comment]
Dumbing it down works. I only use this script as main

```python
from fastapi import FastAPI

app = FastAPI(
    root_path='/api/v1',
    servers=[{'url': '/api/v1'}]
)

@app.get('/')
async def hello():

    return {'result': 'Hello, World!'}
```

My real API using this parameter for app:
* `title`
* `description`
* `openapi_tags` value as list of dictionary
* `default_response_class` value as `ORJSONResponse`
* `responses` to set model for `504 status code`

I also add `FastAPILimiter`. and custom the openapi using this

```python
app.openapi = custom_openapi(app)
```

where `custom_openapi` is this

```python
def custom_openapi(app: FastAPI):
    def wrapper():
        if app.openapi_schema:
            return app.openapi_schema
        openapi_schema = get_openapi(
            title=app.title,
            description=app.description,
            version=app.version,
            routes=app.routes,
            tags=app.openapi_tags
        )

        http_methods = ["post", "get", "put", "delete"]
        # look for the error 422 and removes it
        for method in openapi_schema["paths"]:
            for m in http_methods:
                try:
                    del openapi_schema["paths"][method][m]["responses"]["422"]
                except KeyError:
                    pass
                # try:
                #     if openapi_schema["paths"][method][m]["responses"]["200"]['description'] == 'Successful Response':
                #         del openapi_schema["paths"][method][m]["responses"]["200"]
                # except KeyError:
                #     pass
        for schema in list(openapi_schema["components"]["schemas"]):
            if schema in ["HTTPValidationError", "ValidationError"]:
                del openapi_schema["components"]["schemas"][schema]

        app.openapi_schema = openapi_schema
        return app.openapi_schema
    return wrapper
```

**Comment by fahadh4ilyas** [comment, marked as answer]
Ah, I get it now. This is because `custom_openapi`. I have to add `servers=app.servers` inside `get_openapi`.
